### ISSUE TITLE
refactoring beaqle.js

### ISSUE THREAD

**Issue by nise** [HSU-ANT/beaqlejs#14]
The file beaqle.js has more then 1400 lines of code and contains about 53 functions. This way too much. I would recommend to refactor the code to make it more clear and easy to extend. Currently there are many redundant functions.

At least each type of test should be stored in a separate file. From there you can call interfaces methods provided by the core of beaqle. The core should have multiple components such as the Audiopool, utils, and the listening test among other.

**Comment by seebk** [comment]
Yes, that would make sense as the file has grown bigger and bigger.

**Comment by timobaumann** [comment]
pull request greatly appreciated.

**Comment by nise** [comment]
I will see what I can do. So fare I've simplified a few functions and features that I did not need for my own experiments.
